lib/event-bus.js

```diff
@@ -0,0 +1,61 @@
+const Redis = require('ioredis')
+const EventEmitter = require('events')
+
+/**
+ * This class extends the EventEmitter to act as a pub/sub message bus,
+ * allowing payloads to be received by one instance of Smee and sent by
+ * others. This allows Smee to be multi-instanced!
+ */
+module.exports = class EventBus {
+  constructor () {
+    this.events = new EventEmitter()
+
+    // If Redis isn't enabled, don't try to connect
+    if (!process.env.REDIS_URL) return
+
+    this.opts = {
+      connection: process.env.REDIS_URL,
+      namespace: 'global'
+    }
+
+    // Need two Redis clients; one cannot subscribe and publish.
+    this.sub = new Redis(this.opts.connection)
+    this.pub = new Redis(this.opts.connection)
+
+    // Subscribe to the Redis event channel
+    this.sub.subscribe(this.opts.namespace)
```

Comment:
None of the subscribe() calls in this PR are passing a callback, so any error here will be lost. It could be a good idea to wrap all those calls into a promise and await them.

Reply:
Thank you for taking a look here @sgrondin! Hm according to the type definitions of `ioredis`, `.subscribe` doesn't return a Promise 🤔 I'll do some testing there and try it out anyway.

Reply:
Yeah, the docs don't mention a promise, but I know from experience that it takes a callback as last argument. If it does return a promise then that's great. But if it doesn't you can just wrap the call in `new Promise((resolve, reject) => ...)` and `await` that.